elk/promptsource/templates.py

```diff
@@ -192,9 +192,37 @@ def apply(self, example, truncate=True, highlight_variables=False):
         # Splits on the separator, and then replaces back any occurrences of the
         # separator in the original example
         return [
-            self._unescape_pipe(part).strip() for part in rendered_example.split("|||")
+            Template._strip_spaces(self._unescape_pipe(part))
+            for part in rendered_example.split("|||")
         ]
 
+    @staticmethod
+    def _strip_spaces(string):
```

Comment:
ChatGPT suggests a better way to do this: still use `str.strip()` but simply specify all ASCII space characters except `\n`:
`ascii_whitespace_chars = [chr(i) for i in range(128) if chr(i).isspace() and chr(i) != '\n']`

Reply:
Or simply hardcode them: `['\t', '\x0b', '\x0c', '\r', '\x1c', '\x1d', '\x1e', '\x1f', ' ']`.

Reply:
I think my code is doing something subtly different from that: It is stripping off all the default whitespace characters from both ends, but adding back any of then that were newline characters (so "\n   hi" becomes "\nhi"). I think reframing this way gives me a slightly less clunky way to write the method--I can do that.

Reply:
I should change the docstring then too.

Reply:
I'm just going to go ahead and merge this because it's urgent that we have code which can handle these datasets but we should fix the templates ASAP